src/main/java/edu/cornell/kfs/coa/service/impl/GlobalObjectWithIndirectCostRecoveryAccountsServiceImpl.java

```diff
@@ -0,0 +1,116 @@
+package edu.cornell.kfs.coa.service.impl;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.kuali.kfs.coa.businessobject.IndirectCostRecoveryAccount;
+import org.kuali.rice.krad.bo.GlobalBusinessObjectDetailBase;
+
+import edu.cornell.kfs.coa.businessobject.GlobalObjectWithIndirectCostRecoveryAccounts;
+import edu.cornell.kfs.coa.businessobject.IndirectCostRecoveryAccountChange;
+import edu.cornell.kfs.coa.service.GlobalObjectWithIndirectCostRecoveryAccountsService;
+
+public class GlobalObjectWithIndirectCostRecoveryAccountsServiceImpl implements GlobalObjectWithIndirectCostRecoveryAccountsService{
+	
+	public List<IndirectCostRecoveryAccountChange> getActiveIndirectCostRecoveryAccounts(
+			GlobalObjectWithIndirectCostRecoveryAccounts globalObjectWithIndirectCostRecoveryAccounts) {
+        List<IndirectCostRecoveryAccountChange> activeList = new ArrayList<IndirectCostRecoveryAccountChange>();
+        for (IndirectCostRecoveryAccountChange icr : globalObjectWithIndirectCostRecoveryAccounts.getIndirectCostRecoveryAccounts()){
+            if (icr.isActive()){
+                activeList.add(IndirectCostRecoveryAccountChange.copyICRAccount(icr));
+            }
+        }
+        return activeList;
+    }
+    
+	public void updateIcrAccounts(
+			GlobalObjectWithIndirectCostRecoveryAccounts globalObjectWithIndirectCostRecoveryAccounts,
+			GlobalBusinessObjectDetailBase globalDetail,
+			List<IndirectCostRecoveryAccount> icrAccounts) {
+    	List<IndirectCostRecoveryAccount> updatedIcrAccounts = buildUpdatedIcrAccounts(globalObjectWithIndirectCostRecoveryAccounts, globalDetail, icrAccounts);
+    	globalObjectWithIndirectCostRecoveryAccounts.updateGlobalDetailICRAccountCollection(globalDetail, updatedIcrAccounts);
+    }
+    
+	public List<IndirectCostRecoveryAccount> buildUpdatedIcrAccounts(
```

Comment:
The search logic in this method is similar to that from the "buildCurrentActiveICRAccountsAfterICRUpdates" method in this PR's GlobalIndirectCostRecoveryAccountsRule class. Is there some way that the search logic could be consolidated into one common spot, or are there enough differences between the two to make that impractical?